### PR TITLE
Add first working solution

### PR DESCRIPTION
**Pull request by ondrejsevcik**
TODO:

- [ ] CodeName as constant or attribute - don't know what's better for intended usage
- [ ] Create one executable file that could be easily distributed
- [ ] Improve README
- [ ] Add CI integration
- [ ] Improve logging - I've added very simple logging into the console, maybe that's enough ❔

### CODE REVIEW

File: KenticoCloudDotNetGenerators/CodeGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                    // TODO: not sure what to do with url slug.
+                    continue;
+                }
+
+                try
+                {
+                    var property = Property.FromContentType(element.Codename, element.Type);
+                    classDefinition.AddProperty(property);
+                }
+                catch (InvalidOperationException e)
```
Comment (petrsvihlik) [inline KenticoCloudDotNetGenerators/CodeGenerator.cs]: solve build warnings

**Comment by petrsvihlik** [comment]
Thx Ondra, it works great and the code looks fine. Please check out my review notes + the following remarks:

ad your remarks:
- CodeName - let's solve it in the next iteration when we get to use the metadata (the same goes for url_slug)
- Create one executable - I'll set up appveyor after merging this PR
- Improve README - will do that after merging this PR
- Add CI integration - ditto
- Improve logging - not necessary, CLI logging is just enough

general feedback:
- any reason why you decided to go with .net 4.5 and not .net core?
  - afaik, the referenced libraries support .net standard...
  - the sdk supports .net standard so I guess we should make this generator run on other platforms as well (can do that later in a separate task)
  - we could use Microsoft's https://www.nuget.org/packages/Microsoft.Extensions.CommandLineUtils/

File: KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System.IO;
+using System.Linq;
+
+namespace KenticoCloudDotNetGenerators.Tests
+{
+    [TestFixture]
+    public class CodeGeneratorTests
+    {
+        [TestCase]
+        public void IntegrationTest()
```
Comment (ondrejsevcik) [inline KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs]: We should add way to test DeliveryClient without making actual HTTP requests to the delivery API. That project I'm referencing in that test is my Sandbox project where I do all kinds of experiments, so I can't ensure that content type will stay there forever.

Comment (petrsvihlik) [inline KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs]: right. would it make sense to point it to the same project we reference from the SDK? (or apiary)?

Comment (ondrejsevcik) [inline KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs]: The SDK references my Sandbox project as well so there is the same issue 😄

Comment (petrsvihlik) [inline KenticoCloudDotNetGenerators.Tests/CodeGeneratorTests.cs]: ok, see that now. part of the tests use your project. 

we should use this one: 975bf280-fd91-488c-994c-2f04416e5ee3

**Comment by ondrejsevcik** [comment]
Good point, I didn't think about that. I will convert it to .NET core project.

**Comment by ondrejsevcik** [comment]
I rewrote it to .NET Core. I hope it's what you've asked for :)
I also changed the command line parser to Microsoft's one because CommandLineParser wasn't compatible with .NET Core. 

If you find anything else, just let me know ;)

Thank you

**Comment by petrsvihlik** [comment]
thanks @ondrejsevcik . I made a few changes regarding the project structure and running self-contained on various platforms

**Pull request merged**